odoo/orm/models.py

```diff
@@ -739,11 +740,6 @@ def _build_model_attributes(cls, pool):
 
             for mname, fnames in base._depends.items():
                 depends.setdefault(mname, []).extend(fnames)
-        # copy _sql_constraints into _local_table_objects
-        for cons in cls._sql_constraints:
-            cons_name, definition, cons_message = cons
-            constraint = Constraint(definition, cons_message)
-            constraint.__set_name__(cls, '_' + cons_name)
```

Comment:
Please keep it backward-compatible, there is no reason to drop the support right away.

Reply:
Why? I see reasons to drop it: avoid supporting 2 ways of defining constraints and the migration script is there.
In case we keep it (just removing the last commit): how long should we keep it, should we warn about usage of `_sql_constraints`, what are the reasons to keep it?